docs/extra.rst

```diff
@@ -0,0 +1,168 @@
+Extra
+=====
+
+Chained select2
+---------------
+
+Suppose you have an address form where user should choose a Country and a City.
+The problem is different countries have different cities.
+So we need to show to the user only cities available in the selected country.
+
+Models
+``````
+
+Here are our two models:
+
+.. code-block:: python
+
+    class Country(models.Model):
+        name = models.CharField(max_length=255)
+
+
+    class City(models.Model):
+        name = models.CharField(max_length=255)
+        country = models.ForeignKey('Country')
+
+
+Customizing a Form
+``````````````````
+
+Lets link two widgets via *country_selector* (it's an arbitrary name for CSS class).
+
+.. code-block:: python
+    :emphasize-lines: 7,15,18
+
+    class AddressForm(forms.Form):
+        country = forms.ModelChoiceField(
+            queryset=Country.objects.all(),
+            label=u"Country",
+            widget=ModelSelect2Widget(
+                search_fields=['name__icontains'],
+                attrs={
+                    'class': 'country_selector'
+                }
+            )
+        )
+
+        city = forms.ModelChoiceField(
+            queryset=City.objects.all(),
+            label=u"City",
+            widget=ModelSelect2Widget(
+                data_url='/cities_by_country/',
+                search_fields=['name__icontains'],
+                attrs={
+                    'data-select2-parents': 'country_selector'
+                }
+            )
+        )
+
+
+Registering a view
+``````````````````
+
+Add new route to *urls.py* for retrieving filtered cities:
+
+.. code-block:: python
+
+    from django.conf.urls import patterns, url
+    urlpatterns = patterns(
+        # ...
+        url('^cities_by_country/$', 'cities_by_country', name='cities_by_country'),
+    )
+
+Creating a view
+```````````````
+
+Add a method to *views.py*:
+
+.. code-block:: python
+
+    def cities_by_country(request):
```

Comment:
You could avoid creating your own view, if you would get the `select2-parents` in the view form the widget instance cache.